### PR TITLE
fix(gms): Fixes delete references for single relationship aspects

### PR DESCRIPTION
**Pull request by pedro93**
Deleting Container references were broken. This PR fixes that.

Root cause is https://github.com/datahub-project/datahub/pull/5248 which changed the MCP change type to DELETE for cases where an aspect after getting the reference removed was no longer valid. We do not support  DELETE operations for MCPs.

The solution is then to execute EntityService#deleteAspect rather than emit a sync MCP.

A regression test was made to prevent this from happening again.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by mayurinehate** [comment]
Closes https://github.com/datahub-project/datahub/issues/5992

File: metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java [hunk 1]
```diff
@@ -247,11 +248,10 @@ private void deleteAspect(Urn urn, String aspectName, RecordTemplate prevAspect)
         proposal.setEntityType(urn.getEntityType());
         proposal.setAspectName(aspectName);
 
-        final AuditStamp auditStamp = new AuditStamp().setActor(UrnUtils.getUrn(Constants.SYSTEM_ACTOR)).setTime(System.currentTimeMillis());
-        final EntityService.IngestProposalResult ingestProposalResult = _entityService.ingestProposal(proposal, auditStamp, false);
-
-        if (!ingestProposalResult.isDidUpdate()) {
-            log.error("Failed to ingest aspect with references removed. Before {}, after: null, please check MCP processor"
+        RollbackResult rollbackResult = _entityService.deleteAspect(urn.toString(), aspectName, new HashMap<>(), true);
+        assert rollbackResult != null;
```
Comment (RyanHolstien) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java:252]: Assert statement should get wrapped into below conditional and proposal is no longer used except for the error message which is invalid as it's not being used.

**Review by RyanHolstien — CHANGES_REQUESTED** [review]

**Review by RyanHolstien — APPROVED** [review]

File: metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java [hunk 2]
```diff
@@ -403,6 +396,7 @@ private enum DeleteEntityServiceErrorReason {
         ENTITY_SERVICE_ASPECT_NOT_FOUND,
         ENTITY_REGISTRY_SPEC_NOT_FOUND,
         MCP_PROCESSOR_FAILED,
+        ASPECT_DELETE_FAILED,
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/DeleteEntityService.java:399]: Amazing!

File: metadata-io/src/test/java/com/linkedin/metadata/entity/DeleteEntityServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.Map;
+import org.mockito.Mockito;
+import org.testng.annotations.Test;
+
+import static com.linkedin.metadata.search.utils.QueryUtils.*;
+import static org.mockito.Mockito.*;
+import static org.testng.AssertJUnit.*;
+
+
+public class DeleteEntityServiceTest {
```
Comment (jjoyce0510) [inline metadata-io/src/test/java/com/linkedin/metadata/entity/DeleteEntityServiceTest.java:34]: scary if we did not have a class test for this previously :o

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**